src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -97,20 +96,29 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         _ApplyUISettings();
     }
 
+    // Method Description:
+    // - Funnels focus from TermControl to its input control
+    // Arguments:
+    // - e: the event args for the focus event
+    void TermControl::OnGotFocus(const RoutedEventArgs& e)
+    {
+        if (e.OriginalSource() == *this)
+        {
+            // If we're about to get focus, funnel it into our input site.
+            TerminalInputTarget().Focus(FocusState::Programmatic);
+        }
+    }

```

Comment:
So, this means that whenever the TermControl is attempted to get focus, we redirect focus to the TerminalInputTarget right? So you can't ever accidentally give focus to the search box?

Reply:
Also, just to be clear, this is called just before the `_GotFocusHandler`?

Should we consolidate styles and rename this function to `OnGotFocusHandler`?

Reply:
`OnGotFocus` is a special method name in that the UserControl composition class will call it if it's implemented. It must be named this: it is not a handler, it is not bound to an event.

Reply:
If the _TermControl itself_ attempts to receive focus (not the controls inside it, not the search box, not anything else), we redirect focus to the input target.